mantidproject/mantid
# Move interfaces out of scripts folder

**Description of work.**
This PR moves the interfaces out the scripts folder. There are too many changes to look at with the file changed view, so it will be easier to check the commits.
Implements the design from #32369 
```
├── qt/python
│   ├── mantidqt (this folder contains our mantidqt python package the folder name is not important)
│          └── setup.py # setup file for our mantidqt project
│          └── mantidqt # folder containing mantdqt sources
|                  └── __init__.py 
|                  └── ....
│   ├── mantidqtinterfaces (this folder contains our mantid interfaces python package - the folder name is not important)
│          └── setup.py # setup file for our mantidqtinterfaces project
│          └── mantidqtinterfaces  # folder containing mantid_interfaces sources
│                  └── __init__.py 
|                  └── ....
```

Another change is the mantid.properties file no longer has the location to the interface launch scripts. Instead it is found by checking the module path (to mantidqtinterfaces).

Currently there are two interfaces which haven't been moved:

- Sans (More complex folder structure, not straightforward which things should be pulled out). I don't want to move things about and have the changes un-reviewable (due to how many changes are already in this PR).



**To test:**

- Builds and tests pass
- Checkout the PR. The folders of interest are:
  - qt/python/mantidqtinterfaces
  - scripts/
  - qt/python/mantidqt
- Check the files in these folders seem logical
- Check interfaces open.



Fixes #32369  


*This does not require release notes* because it is an internal change




---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
